### PR TITLE
Support PostgreSQL Custom Format

### PR DESCRIPTION
**Pull request by blackstar257**
##### SUMMARY

postgresql_db: add support for `.pgc` file format for dump and restores

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

`lib/ansible/modules/database/postgresql/postgresql_db.py`

### CODE REVIEW

File: lib/ansible/modules/database/postgresql/postgresql_db.py [hunk 1]
```diff
@@ -391,6 +391,10 @@ def db_restore(module, target, target_opts="",
     elif os.path.splitext(target)[-1] == '.tar':
         flags.append(' --format=Tar')
         cmd = module.get_bin_path('pg_restore', True)
+
```
Comment (tcraxs) [inline lib/ansible/modules/database/postgresql/postgresql_db.py]: Please remove the trailing whitespace here:
`blank line contains whitespace`

**Review by tcraxs — CHANGES_REQUESTED** [review]

**Comment by tcraxs** [comment]
Thanks @blackstar257 for your PR. Can you please add a test case for that new feature. Also a changelog entry would be nice: https://docs.ansible.com/ansible/latest/community/development_process.html#changelogs

For an example of a changelog see here: #63555

**Review by Andersson007 — CHANGES_REQUESTED** [review]
@blackstar257 , hey, thanks for working on this! As @tcraxs said we need to have CI for that:
1. test case with check_mode: yes with assertions, etc., see ```test/integration/targets/postgresql_db/tasks/``` 
2. test case without check mode

Also, yep, the changelog fragment is required for fixes like this

**Comment by blackstar257** [comment]
Will do ill tackle it when i get a chance.

File: lib/ansible/modules/database/postgresql/postgresql_db.py (hunk 1 above)
Comment (blackstar257) [reply lib/ansible/modules/database/postgresql/postgresql_db.py]: Fixed

**Comment by blackstar257** [comment]
@Andersson007 @tcraxs i believe i made all the appropriate changes

**Review by Andersson007 — APPROVED** [review]
LGTM, thanks for the contribution

shipit

**Comment by Dorn-** [comment]
Gj.
shipit

**Pull request merged**